src/v2/Apps/ViewingRoom/ViewingRoomApp.tsx

```diff
@@ -20,26 +24,44 @@ const ViewingRoomApp: React.FC<ViewingRoomAppProps> = ({
   children,
   viewingRoom,
 }) => {
+  const { mediator, user } = useContext(SystemContext)
+  useEffect(() => {
+    if (!user || !user.id) {
+      openAuthModal(mediator, {
+        mode: ModalType.signup,
+        redirectTo: window.location.href,
+        contextModule: ContextModule.viewingRoom,
+        intent: Intent.viewViewingRoom,
+      })
+    }
+  }, [user, mediator])
+
   if (!viewingRoom) {
     return <ErrorPage code={404} />
   }
 
+  const getView = () => {
+    if (!user) {
+      return
+    } else if (viewingRoom.status === "closed") {
+      return <ViewingRoomClosed viewingRoom={viewingRoom} />
+    } else {
+      return (
+        <>
+          <ViewingRoomTabBar mb={[2, 3]} />
+          {children}
+        </>
+      )
+    }
+  }
```

Comment:
I'd probably extract this into an actual component rather than a scoped function like this, if it can be named appropriately — primarily because I'd imagine there should be some kind of UI/message/CTA to prompt login if the user closes the auth modal?

Reply:
Also - should even the `Header` be displayed in that case?

Should this logic come one level up? Like similar to how we are returning if there is no `ViewingRoom` ?

Reply:
@dzucconi ah that makes sense, since this is a bit time-sensitive, I'll probably ticket that change and come back to it later. Will also check with design on any messaging that should happen when the user closes the auth modal.

@oxaudo I think the design is to have the header in the background as a teaser with a the signup modal overlaid. Definitely is a weird half state to show, but I think it makes sense.

Reply:
Ah.. I see. Ok:)